src/tcgwars/logic/impl/gen7/ForbiddenLight.groovy

```diff
@@ -2590,13 +2590,22 @@ public enum ForbiddenLight implements LogicCardInfo {
       case DIANTHA_105:
         return supporter (this) {
           text "You can play this card only if 1 of your [Y] Pokémon was Knocked Out during your opponent’s last turn.\nPut 2 cards from your discard pile into your hand.\nYou may play only 1 Supporter card during your turn (before your attack)."
+          globalAbility {Card thisCard->
+            delayed {
+              before KNOCKOUT, {
+                if(ef.pokemonToBeKnockedOut.types.contains(Y) && ef.pokemonToBeKnockedOut.owner == thisCard.player && bg.currentTurn == thisCard.player.opposite){
+                  keyStore("Diantha_KO", thisCard, bg.turnCount)
+                }
+              }
+            }
+          }
           onPlay {
             my.discard.select(count : Math.min(2,my.discard.size())).moveTo(my.hand)
           }
           playRequirement{
+            assert my.discard : "You have no cards in your discard pile"
             assert bg.turnCount
-            assert my.lastKnockoutByOpponentDamageTurn == bg.turnCount - 1: "No Pokémon has been Knocked Out during your opponent’s last turn"
-            assert my.knockoutTypesPerTurn.get(bg.turnCount - 1)?.contains(Y) : "The Pokémon Knocked Out was not Fairy"
```

Comment:
So sad that this doesn't work as `knockoutTypesPerTurn` structure was specifically created for this card 😅 Maybe it's better to update `CheckFaint` . Nevertheless, it can be postponed if you think there is not enough time for now.

Reply:
Yeah, there are many cards that are heavily played and have this kind of fix. It can definitely be moved over those, but would rather leave it for a separate PR 😅